hack/prebuild/init.sh

```diff
@@ -0,0 +1,19 @@
+#/bin/bash
+
+yum install -y -d1 libvirt libvirt-daemon-kvm
+usermod -aG libvirt root
+
+cat <<EOF > /etc/libvirt/libvirtd.conf
+unix_sock_group = "libvirt"
+unix_sock_rw_perms = "0770"
+listen_tls = 0
+listen_tcp = 1
+auth_tcp="none"
+tcp_port = "16509"
+EOF
+
+echo 'LIBVIRTD_ARGS="--listen"' >> /etc/sysconfig/libvirtd
+
+iptables -I INPUT -p tcp --dport 16509 -j ACCEPT -m comment --comment "Allow insecure libvirt clients"
```

Comment:
Are the tests going to use TCP or SSH as a transport for libvirt? I don't think it's okay to run these with unauthenticated libvirt listening on the public internet.

Reply:
For CI I was thinking on parameterising the ip to allowed to access and then reuse your tls stuff, so we test it as it's consumed by the installer/mao. wdyt?

Reply:
Yeah, that sounds good. I'm not familiar with the e2e tests or the plan for those. Do they use the installer? If so, I think this should be on-hold until https://github.com/openshift/installer/pull/296 merges.

Reply:
Eventually we might use installer. Currently for mao/aws-actuator we use terraform to create the minimal infra needed to satisfy some actuator assumptions and verify expected behaviour, 
See https://github.com/openshift/machine-api-operator/tree/master/tests
e.g The minimal infra for libvirt actuator will contain a network and the ign volumes

Reply:
Ah, cool. Then I think it should be pretty easy to use TLS or SSH as the transport from the start. There aren't really any changes required for the actuator. The credentials, certs or SSH keys, just need to be available and the right URI needs to be used. The libvirt library will handle the rest.

Reply:
Just to add a little more detail on making TLS work: [here's a link to the docs](https://github.com/openshift/installer/blob/41a24535cb6025827ff4976650424d439e8c6661/docs/dev/libvirt-howto.md#17-configure-libvirt-to-accept-tls-connections)

We just need to generate the TLS assets somehow -- could use certtool, cfssl, or even Terraform if we don't want to use the Go tool in the installer PR. Then configure libvirtd to point to the certs and use a URI like: `qemu+tls://$IP/system?pkipath=/path/to/certs`

That should be enough and pretty secure as long as no one can access the generated certs and they are one-time use.

Reply:
It would be probably easier to adapt it to use SSH as this already need SSH keys to connect to instance.

Reply:
Totally fine. As long as we're doing authentication and encryption somehow.